JASP-Engine/JASP/R/descriptives.R

```diff
@@ -111,6 +111,47 @@ Descriptives <- function(jaspResults, dataset, options) {
       }
     }
   }
+
+  # QQ plots
+  if (options$descriptivesQQPlot) {
+    if(is.null(jaspResults[["QQPlots"]])) {
+      if(length(variables)>1 || length(levels(dataset[[.v(splitName)]]))>1) #there will be more than one Q-Q Plot
+        jaspResults[["QQPlots"]] <- createJaspContainer("Q-Q Plots")
+      else #only one Q-Q Plot
+        jaspResults[["QQPlots"]] <- createJaspContainer("Q-Q Plot")
+      jaspResults[["QQPlots"]]$dependOn(c("descriptivesQQPlot", "variables", "splitby"))
+      jaspResults[["QQPlots"]]$position <- 8
+    }
+    QQPlots <- jaspResults[["QQPlots"]]
+    if(makeSplit) {
+      qqSplitFactor     <- dataset[[.v(splitName)]]
+      if(length(qqSplitFactor)==0)
+        return(createJaspPlot(error="Plotting is not possible: Variable only contains NA!", dependencies="splitby"))
+      #gives the different split values
+      qqSplitLevels     <- levels(qqSplitFactor)
+      # remove missing values from the grouping variable
+      dataset           <- dataset[!is.na(qqSplitFactor), ]
+      for(var in variables){ 
+        deeperQQPlots <- createJaspContainer(paste0(var))
+        QQPlots[[var]] <- deeperQQPlots
+        #splits dataset according to split values
+        qqSplitData     <- split(dataset, qqSplitFactor)
+        for( lev in 1:length(qqSplitLevels)){
+          QQPlots[[var]][[paste0(var, lev)]] <- .descriptivesQQPlot(dataset=qqSplitData[[lev]], options=options, qqvar=var, levelName=qqSplitLevels[lev])
+        }
+      }
+    }
+    else { #no split
+      for(var in variables){
+        if(is.null(QQPlots[[var]])) {
```

Comment:
This will always be `NULL` right now, unless you change the dependencies as suggested, at which point this becomes a useful if-statement